build.sh

```diff
@@ -5,80 +5,106 @@
+# because grep exits with an error code, causing the pipeline to fail, script exit
+#
+# a workaround is to enclose grep as follows, where e is search term:
+#
+# cmd | { grep e || :; } | cmd
+#
+set -eo pipefail
+
+exit_error() {
+    echo "$*" 1>&2
+    exit 1;
+}
+
+usage() {
+    cat <<-EOF
+	Usage: $(basename "${BASH_SOURCE[0]}") [-hu] [-k /path/to/authorized_keys] 
+	
+	-h Show help
+	-u Build with upload to remote host
+	-k /path/to/authorized_keys (can safely ignore, another opportunity to copy in SSH keys on image boot!)
+	EOF
+}
+
 # we must be on freebsd
 what_os_am_i="$(uname)"
 if [ "${what_os_am_i}" != "FreeBSD" ]; then
-    echo "Please run on FreeBSD only"
-    exit
-fi
-
-if [ $# -lt 1 ]; then
-  1>&2 echo "
-Usage: $0 -i access-ip -f /path/to/authorized_keys -u [0/1]
-
-       Example with upload on:
-       $0 -i \"10.0.0.1\" -f /root/copy_in_auth_keys -u 1
-"
-  exit 1
+    exit_error "Please run on FreeBSD only"
 fi
 
+UPLOAD="NO"
 # get command line flags
-while getopts i:f:u: flag
+while getopts huk: flag
 do
     case "${flag}" in
-        i) MYACCESSIP="${OPTARG}";;
-        f) AUTHKEYFILE="${OPTARG}";;
-        u) UPLOAD="${OPTARG}";;
+        h) 
+           usage
+           exit 0
+           ;;
+        u) 
+           UPLOAD="YES"
+           ;;
+        k) 
+           AUTHKEYFILE="${OPTARG}"
+           ;;
+        *) 
+           exit_error "$(usage)"
+           ;;
     esac
 done
+shift "$((OPTIND-1))"
 
-# VARS
-if [ -z "${MYACCESSIP}" ]; then
-    MYACCESSIP="*"
-fi
+# Set default values if not set
+# XXX: this might be unnecessary
 if [ -z "${AUTHKEYFILE}" ]; then
     touch authorized_keys_in
     AUTHKEYFILE="authorized_keys_in"
 fi
-if [ -z "${UPLOAD}" ]; then
-    UPLOAD=0
-fi
 
-# VARS
+# General Variables
 BASEDIR="$PWD"
 CDMOUNT="cd-rom"
-CHECKMOUNTCD1=$(mount | grep cd-rom | awk -F" " '{print $1}')
+CHECKMOUNTCD1="$(mount | { grep "${CDMOUNT}" || :; } | awk '{print $1}')"
 FREEBSDISOSRC="https://download.freebsd.org/releases/amd64/amd64/ISO-IMAGES/13.1/FreeBSD-13.1-RELEASE-amd64-dvd1.iso"
 FREEBSDISOFILE="FreeBSD-13.1-RELEASE-amd64-dvd1.iso"
 MFSBSDDIR="mfsbsd"
 MYRELEASE="13.1-RELEASE"
 MYARCH="amd64"
-OUTIMG="mfsbsd-${MYRELEASE}-${MYARCH}.img"
-OUTISO="mfsbsd-${MYRELEASE}-${MYARCH}.iso"
+OUTIMG="mfsbsd-${MYRELEASE}-${MYARCH}.img"    # not in use
+OUTISO="mfsbsd-${MYRELEASE}-${MYARCH}.iso"    # in use
 MYBASE="${BASEDIR}/${CDMOUNT}/usr/freebsd-dist"
-MYREMOTECONFIG="${BASEDIR}/settings.cfg"
 MYCUSTOMDIR="${BASEDIR}/customfiles"
 
 # make sure we're in base directory
 cd "${BASEDIR}" || exit
 
 # check remote settings
-if [ -f "${MYREMOTECONFIG}" ]; then
-    source "${MYREMOTECONFIG}"
+if [ -f "${BASEDIR}/settings.sh" ]; then
+    # shellcheck source=/dev/null
+    source "${BASEDIR}/settings.sh"
 else
-    echo "Please copy settings.cfg.sample to settings.cfg and set remote parameters"
-    exit
+    exit_error "Please copy settings.sh.sample to settings.sh and set parameters"
+fi
```

Comment:
These two lines have to go, so it's possible to pass in `CFG_SSH_REMOTEHOST` / have a default value when no settings.sh was created.

Reply:
I marked the wrong lines, I meant these two:

```
else
    exit_error "Please copy settings.sh.sample to settings.sh and set parameters"
```

Reply:
fixed